### PR TITLE
lndclient: expose ListTransactions as part of LightningClient

### PR DESCRIPTION
**Pull request by wpaulino**

### CODE REVIEW

File: test/lightning_client_mock.go [hunk 1]
```diff
@@ -119,3 +120,9 @@ func (h *mockLightningClient) AddInvoice(ctx context.Context,
 
 	return hash, payReqString, nil
 }
+
+func (h *mockLightningClient) ListTransactions(
+	ctx context.Context) ([]*wire.MsgTx, error) {
+
+	return nil, nil
```
Comment (guggero) [inline test/lightning_client_mock.go]: Would be nice if we could control what the mock returns by adding a variable to `test/lnd_services_mock.go` and returning it here.

Comment (wpaulino) [reply test/lightning_client_mock.go]: Done, also included in this list the transactions by `SendOutputs` and `PublishTransaction`.

File: test/lnd_services_mock.go [hunk 2]
```diff
@@ -122,6 +122,8 @@ type LndMockServices struct {
 
 	Height int32
 
+	RelevantTransactions []*wire.MsgTx
```
Comment (joostjager) [inline test/lnd_services_mock.go]: Is the word `Relevant` accurate in the mock context? I'd expect something like `AllTransactions`

Comment (wpaulino) [reply test/lnd_services_mock.go]: Fixed.

**Pull request merged**